sonar-db/src/main/java/org/sonar/db/issue/ActionPlanDao.java

```diff
@@ -78,23 +78,31 @@ public ActionPlanDto selectByKey(String key) {
   }
 
   public List<ActionPlanDto> selectByKeys(Collection<String> keys) {
-    if (keys.isEmpty()) {
-      return Collections.emptyList();
-    }
-    SqlSession session = mybatis.openSession(false);
+    DbSession session = mybatis.openSession(false);
     try {
-      List<ActionPlanDto> dtosList = newArrayList();
-      List<List<String>> keysPartition = Lists.partition(newArrayList(keys), 1000);
-      for (List<String> partition : keysPartition) {
-        List<ActionPlanDto> dtos = session.getMapper(ActionPlanMapper.class).findByKeys(partition);
-        dtosList.addAll(dtos);
-      }
-      return dtosList;
+      return selectByKeys(session, keys);
     } finally {
       MyBatis.closeQuietly(session);
     }
   }
 
+  public List<ActionPlanDto> selectByKeys(DbSession dbSession, Collection<String> keys) {
+    return DatabaseUtils.executeLargeInputs(keys, new SelectByKeys(dbSession.getMapper(ActionPlanMapper.class)));
+  }
+
+  private static class SelectByKeys implements Function<List<String>, List<ActionPlanDto>> {
```

Comment:
No, this function is specific to this method and isn't/won't be reused elsewhere.